Dockerfile

```diff
@@ -12,14 +12,15 @@ RUN apt-get update && \
 
 WORKDIR /app
 
-# Pin a known to work with peep pip version.
-RUN pip install pip==6.0.0
+# Get pip8
+COPY bin/pipstrap.py bin/pipstrap.py
+RUN ./bin/pipstrap.py
 
 # First copy requirements.txt and peep so we can take advantage of
 # docker caching.
-COPY ./bin/peep.py /app/bin/peep.py
-COPY requirements.txt /app/requirements.txt
-RUN ./bin/peep.py install -r requirements.txt
+COPY requirements requirements
+COPY requirements.txt requirements.txt
+RUN pip install --require-hashes --no-cache-dir -r requirements.txt
```

Comment:
The answer is "size" since pip will be default keep the downloaded packages in cache in the docker image. :)